pulsar-client/src/main/java/org/apache/pulsar/client/api/ProducerConfiguration.java

```diff
@@ -50,11 +49,39 @@
 
     private CompressionType compressionType = CompressionType.NONE;
 
+    // Cannot use Optional<Long> since it's not serializable
+    private Long initialSequenceId = null;
```

Comment:
just a quick question: what is the reason for making `ProducerConfiguration` a `Serializable`?

Reply:
It's already implementing `Serializable` and the previous change was breaking it. The only reason is to support the Storm spout. All the configs get filled in at the topology submission time and then serialized and sent to the worker nodes.

Reply:
oh.. yes, make sense.